### PR TITLE
Refactor Keys model to use PineJS

### PR DESCRIPTION
**Pull request by jviotti**
This PR should be tested and merged once the corresponding backend changes are in production.

### CODE REVIEW

File: lib/models/key.coffee [hunk 1]
```diff
@@ -33,16 +32,23 @@ request = require('resin-request')
...
+	username = token.getUsername()
+
+	if not username?
+		return callback(new errors.ResinNotLoggedIn())
+
+	return pine.get
+		resource: 'user__has__public_key'
+		options:
+			filter:
+				user: { username }
```
Comment (emirotin) [inline lib/models/key.coffee]: why not by user ID?

Comment (jviotti) [inline lib/models/key.coffee]: @emirotin Much better. Do you mind taking another look?

File: lib/models/key.coffee [hunk 2]
```diff
@@ -33,16 +33,21 @@ request = require('resin-request')
...
 		return callback(error) if error?
 
-		if _.isEmpty(keys)
-			return callback(new errors.ResinNotAny('keys'))
+		return pine.get
+			resource: 'user__has__public_key'
+			options:
+				filter:
+					user: { id }
+		.then (keys) ->
```
Comment (Page-) [inline lib/models/key.coffee]: If you use `.tap` then you don't need the `return keys` below

File: lib/models/key.coffee [hunk 3]
```diff
@@ -65,15 +70,22 @@ exports.getAll = (callback) ->
...
+		return pine.get
+			resource: 'user__has__public_key'
+			id: id
+			options:
+				filter:
+					user: { id: userId }
+		.then (key) ->
+			if _.isEmpty(key)
+				throw new errors.ResinKeyNotFound(id)
+			return _.first(key)
```
Comment (Page-) [inline lib/models/key.coffee]: You could replace this with:
```
.tap (key) ->
	if _.isEmpty(key)
		throw new errors.ResinKeyNotFound(id)
.get(0)
```

File: lib/models/key.coffee (hunk 2 above)
Comment (jviotti) [inline lib/models/key.coffee]: Fixed.

File: lib/models/key.coffee (hunk 3 above)
Comment (jviotti) [inline lib/models/key.coffee]: Fixed.

**Comment by Page-** [comment]
LGTM

**Pull request merged**